src/storage/gcloud-storage/gcp-storage.service.unit.test.ts

```diff
@@ -225,4 +252,120 @@ describe('GCPStorageService', () => {
             });
         });
     });
+
+    describe('compressFile', () => {
+        beforeEach(() => {
+            service = new GCPStorageService(mockLogger);
+        });
+
+        it('compresses a JPEG file and reduces its size', async () => {
+            const mockFile = await createMockFile(
+                'image/jpeg',
+                'image.jpg',
+                2000,
+                2000,
+            );
+            const originalSize = mockFile.size;
+
+            const compressedBuffer = await (
+                service as unknown as {
+                    compressFile: (file: File) => Promise<Buffer>;
+                }
+            ).compressFile(mockFile);
```

Comment:
This feels off, but I get why you need to do it (testing privates)

What do you think to moving this image handling to another service, like `ImageService`? This would give us a nice separation of concerns (the storage service specifically deals with writing / reading to storage) and then would make sense for `compressFile` to be on the public interface, making it easier to test

Reply:
Yeah I think that's a good idea! Currently the `gcp-storage-service` is mixing logic that is specific to images (e.g. allowed images types, or storage path with images/ or now the image compression) with logic that is generic to any file storage. 

I suggest: I merge this PR with image compression logic, and then start a follow-up refactoring PR to avoid making this one too big / impossible to review. The refactoring is a good one to look at during clean code sesh too

Reply:
Yeh I didn't look too deeply into what else the storage service is doing, but this approach works for me 👍